python/cuml/common/base.pyx

```diff
@@ -282,6 +282,24 @@ class Base:
             else:
                 raise AttributeError
 
+    def _set_base_attributes(self, X, output_type=False, y=None,
+                             n_features=None):
+        """
+        X: input data
+        output_type: Boolean; if we should set output type on X
+        y: target column, passed if we should set target dtype
+        n_features: Can be Boolean or int. If an int is passed
+                    we set it to the number passed
```

Comment:
It would be nice if the formatting of the docstring matches our regular formatting, and adding one or 2 examples of usage would be good

Reply:
I've added a few examples.